### PR TITLE
Don't report .pyc file writes in sandbox logs

### LINKED ISSUE #683: filter .pyc messages out of sandbox log
_From @tdsmith on January 3, 2016 18:51_

Messages of the form `.*Python(\d+) deny file-write-create.*pyc` are just noise and aren't fatal; it would be nice to hide them from the sandbox log summary.

_Copied from original issue: Homebrew/legacy-homebrew#47640_

### PR DESCRIPTION
**Pull request by tdsmith**
These are never fatal and often confusing.

Fixes #683.

### CODE REVIEW

File: Library/Homebrew/sandbox.rb [hunk 1]
```diff
@@ -95,6 +95,10 @@ def exec(*args)
       -k Sender sandboxd
     ]
     logs = Utils.popen_read("syslog", *syslog_args)
+
+    # These messages are confusing and non-fatal, so don't report them.
+    logs = logs.gsub(/^.*Python(\d+) deny file-write.*pyc$/, "").strip
```
Comment (MikeMcQuaid) [inline Library/Homebrew/sandbox.rb]: Does this strip the newlines too or need a ` \n`?

File: Library/Homebrew/test/test_sandbox.rb [hunk 2]
```diff
@@ -25,4 +25,21 @@ def test_deny_write
...
+  def test_complains_on_failure
+    Utils.expects(:popen_read => "foo")
+    out, err = capture_io do
+      assert_raises(ErrorDuringExecution) { @sandbox.exec "false" }
+    end
+    assert_match out, "foo"
+  end
+
+  def test_ignores_bogus_python_error
+    bogus_error = "Mar 17 02:55:06 sandboxd[342]: Python(49765) deny file-write-unlink /System/Library/Frameworks/Python.framework/Versions/2.7/lib/python2.7/distutils/errors.pyc\n"
```
Comment (MikeMcQuaid) [inline Library/Homebrew/test/test_sandbox.rb]: Maybe worth making it multiline and checking that other lines aren't truncated. I know they aren't from the code above but might be worth making the test a little more robust.

**Comment by MikeMcQuaid** [comment]
Looks good 👍. A few tiny questions/nits but should be good to go soon. Nice work.

File: Library/Homebrew/sandbox.rb (hunk 1 above)
Comment (tdsmith) [inline Library/Homebrew/sandbox.rb]: It'd have to be a multiline regex, it turns out; I don't know why I thought `strip` would fix this. ☕☕☕ Tried an approach with `.lines` instead; let me know how you like it.

File: Library/Homebrew/test/test_sandbox.rb (hunk 2 above)
Comment (tdsmith) [inline Library/Homebrew/test/test_sandbox.rb]: While looking at this, discovered that all my `assert_match`es were backwards, oops. Agree that it improves the test~!

**Comment by MikeMcQuaid** [comment]
LGTM 👍, thanks for the changes.

**Pull request closed without merging**